pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -3246,6 +3237,16 @@ def visit_functiondef(self, node: nodes.FunctionDef) -> None:
                     return
             matchers = self._function_matchers
 
+        # Check that common arguments are correctly typed.
+        for arg_name, expected_type in _COMMON_ARGUMENTS.items():
```

Comment:
Note to the reviewer: this is moved down as it no longer needs to be run for test function or test fixtures (early exit above)